main.cpp

```diff
@@ -227,163 +147,92 @@ struct Numeric
 
     operator Type() const { return *value; }
 
-    Numeric& operator+=(const Type& t)
-    {
-        *value += t;
-        return *this;
+    operator NumericType() const { return *value; }
+    operator NumericType&() { return *value; }
+
+    template<typename OtherType>
+    Numeric& operator=(const OtherType& o) 
+    { 
+        *value = static_cast<NumericType>(o); 
+        return *this; 
     }
 
-    Numeric& operator-=(const Type& t)
+    template<typename OtherType>
+    Numeric& operator+=(const OtherType& o) // #3
     {
-        *value -= t;
+        *value += static_cast<NumericType>(o);
         return *this;
     }
 
-    Numeric& operator*=(const Type& t)
+    template<typename OtherType>
+    Numeric& operator-=(const OtherType& o) // #3
+    { 
+        *value -= static_cast<NumericType>(o); 
+        return *this; 
+    }
+
+    template<typename OtherType>
+    Numeric& operator*=(const OtherType& o) // #3
     {
-        *value *= t;
+        *value *= static_cast<NumericType>(o);
         return *this;
     }
 
-    // #11
-    template<typename DivisorType> 
-    Numeric& operator/=(const DivisorType& t)
+    template<typename OtherType> 
+    Numeric& operator/=(const OtherType& o) // #3
     {
-        // struct Numeric template type is int
-        if constexpr (std::is_same<Type,int>::value)
+        // template type is int
+        if constexpr (std::is_same<OtherType,int>::value)
         {
-            // function parameter type is also int
-            if constexpr (std::is_same<DivisorType,int>::value)
+            // function parameter type is int
+            if constexpr (std::is_same<decltype(o),int>::value)
             {
                 // function parameter is integer 0 don't do the division
-                if (t == 0)
+                if (o == 0)
                 {
                     std::cerr << "can't divide integers by zero!" << std::endl;
                     return *this;
                 }
             }
-            else if ( t < std::numeric_limits<DivisorType>::epsilon() )
+            else if ( o < std::numeric_limits<OtherType>::epsilon() )
             {
                 // else if function parameter is less than epsilon dont do the divison
                 std::cerr << "can't divide integers by zero!" << std::endl;
                 return *this;
             }
         } 
-        else if ( t < std::numeric_limits<Type>::epsilon() )
+        else if ( o < std::numeric_limits<OtherType>::epsilon() )
```

Comment:
template type, not function parameter type.
- [ ] revise

Reply:
Done.